src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -122,11 +124,14 @@ export function CollapsibleNav({
   const recentlyAccessed = useObservable(observables.recentlyAccessed$, []);
   const customNavLink = useObservable(observables.customNavLink$, undefined);
   const appId = useObservable(observables.appId$, '');
+  const currentWorkspace = useObservable(observables.currentWorkspace$);
   const lockRef = useRef<HTMLButtonElement>(null);
   const groupedNavLinks = groupBy(navLinks, (link) => link?.category?.id);
   const { undefined: unknowns = [], ...allCategorizedLinks } = groupedNavLinks;
-  const categoryDictionary = getAllCategories(allCategorizedLinks);
-  const orderedCategories = getOrderedCategories(allCategorizedLinks, categoryDictionary);
+  const filterdLinks = getFilterLinks(currentWorkspace, allCategorizedLinks);
```

Comment:
We may use `useMemo` to optimize the computed / filtered.

Reply:
Thanks for your remind. Here I originally planned to be consistent with the original `getAllCategories` and `getOrderedCategories` function, I can add it in the next commit.